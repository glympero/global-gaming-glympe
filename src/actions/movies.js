//SET_MOVIES
const setMovie = (movies) => ({
    type: 'SET_MOVIES',
    movies
  });

// REMOVE_MOVIE
const removeMovie = ({ id } = {}) => ({
    type: 'REMOVE_MOVIE',
    id
});

// EDIT_MOVIE
const editMovie = (id, updates) => ({
    type: 'EDIT_MOVIE',
    id,
    updates
});

export { setMovie, removeMovie, editMovie };