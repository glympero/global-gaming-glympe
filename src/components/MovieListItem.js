import React from 'react';
import PropTypes from 'prop-types';
import Modal from './Modal';
import MovieEdit from './MovieEdit';
import moment from 'moment';

class MovieListItem extends React.Component {

  constructor(props) {
    super(props);
    this.state = { isOpen: false };
  }

  toggleModal = () => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }
  render() {
    return (
      <div>
        <div className='list-item' onClick={this.toggleModal}>
          <h3 className='list-item__title'>{this.props.title}</h3>
          <p className='list-item__subtitle'>{this.props.genre}</p>
          <p className='list-item__author'>{this.props.director.name}</p>
        </div>
        <Modal show={this.state.isOpen}>
          <MovieEdit id={this.props.id} onClose={this.toggleModal} />
        </Modal>    
      </div>
    );
  }
};

MovieListItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  genre: PropTypes.string.isRequired,
  director: PropTypes.object.isRequired,
  release_date: PropTypes.string.isRequired
};

export default MovieListItem;