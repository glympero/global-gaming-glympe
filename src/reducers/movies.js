const moviesReducerDefaultState = [];

const moviesReducer = (state = moviesReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_MOVIES':
            return action.movies;
        case 'REMOVE_MOVIE':
            return state.filter(movie => movie.id !== action.id);
        case 'EDIT_MOVIE':
            return state.map(movie => {
                if(movie.id === action.id) {
                    return {
                        ...movie,
                        ...action.updates
                    }
                }
                return movie;
            })
        default:
            return state;
    }
};

export default moviesReducer;