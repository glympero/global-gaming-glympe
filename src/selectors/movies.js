const getVisibleMovies = (movies, filters) => {
    const {text, sortBy, genre } = filters;
    return movies.filter(movie => {
        const textMatched = typeof text === 'string' && movie.title.toLowerCase().indexOf(text.toLowerCase()) !== -1;
        const genreMatched = typeof genre === 'string' && genre === 'all' ? true : movie.genre.toLowerCase().indexOf(genre.toLowerCase()) !== -1;
        return textMatched && genreMatched;
    }).sort((a, b) => {
        if(sortBy === 'title') {
            return a.title > b.title ? 1 : -1;
        }else if (sortBy === 'director') {
            return a.director.toLowerCase() > b.director.toLowerCase() ? 1 : -1;
        }
    });
};

export default getVisibleMovies;