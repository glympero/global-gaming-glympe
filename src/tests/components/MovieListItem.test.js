import React from 'react';
import { shallow } from 'enzyme';
import MovieListItem from '../../components/MovieListItem';
import movies from '../../fixtures/data';

test('should render MovieListItem with props', () => {
    const wrapper = shallow(<MovieListItem key={movies[0].id} {...movies[0]} />)
    expect(wrapper).toMatchSnapshot();
})