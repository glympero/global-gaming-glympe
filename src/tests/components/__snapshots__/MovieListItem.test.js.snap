// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieListItem with props 1`] = `
<div>
  <div
    className="list-item"
    onClick={[Function]}
  >
    <h3
      className="list-item__title"
    >
      Absolute Bite
    </h3>
    <p
      className="list-item__subtitle"
    >
      Drama@april.biz
    </p>
    <p
      className="list-item__author"
    >
      Walsh Mertie
    </p>
  </div>
  <Modal
    show={false}
  >
    <Connect(MovieEdit)
      id="1"
      onClose={[Function]}
    />
  </Modal>
</div>
`;
